### ISSUE TITLE
V2: Service Error Responses Broken

### ISSUE THREAD

**Issue by sachnk** [goadesign/goa#1905]
Hey guys,

I'm struggling with getting service errors (like validation errors from missing fields, etc) encoded to a proper response with status codes in the latest V2 branch. 

## Expected Behavior

Running `goa.design/goa/examples/calc` example, and doing a `GET` on `/add/1/foobar` should return a properly encoded `400 BadRequest` since the "foobar" is not an integer.

## Actual Behavior

`GET` on `/add/1/foobar` actually returns a non-encoded payload with a `200 OK` response.

## Steps to Reproduce the Problem

Simply run the provided calc example as described above.

## Specifictions

- goa Version: V2 @ 702c321d0f86bcdce8c2f36a4bf4b2f21230dc19
- Platform: Mac

I can actually fix the calc example by modifying the `ErrorHandler` code in `main.go` to the following:

```golang
func ErrorHandler(logger *log.Logger) func(context.Context, http.ResponseWriter, error) {
	encodeError := goahttp.ErrorEncoder(goahttp.ResponseEncoder)  // this line added
	return func(ctx context.Context, w http.ResponseWriter, err error) {
		id := ctx.Value(middleware.RequestIDKey).(string)
		logger.Printf("[%s] ERROR: %s", id, err.Error())
		encodeError(ctx, w, err) // this line added, to encode the response payload
	}
}
```

But I'm not sure if this the idiomatic way to fix this...

**Comment by raphael** [comment]
Thank you, this looks like a bug.

**Comment by raphael** [comment]
Thank you for the detailed report, this is fixed in #1917
